sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -311,4 +322,15 @@ public AmazonCloudWatch getCloudWatchClient() {
       }
     }
   }
+
+  static boolean streamExists(AmazonKinesis client, String streamName) {
+    try {
+      DescribeStreamResult describeStreamResult = client.describeStream(streamName);
+      StreamDescription streamDescription = describeStreamResult.getStreamDescription();
+      return streamName.equals(streamDescription.getStreamName());
```

Comment:
Is there any way this can return false?

Reply:
Well that's right that validation was quite silly. From looking at the official doc on the response of the Aws Kinesis API it actually is status code 200 or exception, so I updated to do it like that, for future ref.
https://docs.aws.amazon.com/kinesis/latest/APIReference/API_DescribeStream.html

I hope everything is ok now, thanks again and sorry for the silly validation.